keras_cv/models/backbones/resnet_v2/resnet_v2_backbone.py

```diff
@@ -482,189 +364,223 @@ def get_config(self):
-        stackwise_strides=MODEL_CONFIGS["ResNet50V2"]["stackwise_strides"],
-        include_rescaling=include_rescaling,
-        include_top=include_top,
-        name=name,
-        weights=parse_weights(weights, include_top, "resnet50v2"),
-        input_shape=input_shape,
-        input_tensor=input_tensor,
-        pooling=pooling,
-        classes=classes,
-        classifier_activation=classifier_activation,
-        block_type="block",
+    @classproperty
+    def presets_with_weights(cls):
+        """Dictionary of preset names and configurations that include weights."""
+        return {}
+
+
+class ResNet34V2Backbone(ResNetV2Backbone):
+    def __new__(
+        self,
+        include_rescaling=True,
+        input_shape=(None, None, 3),
+        input_tensor=None,
         **kwargs,
-    )
+    ):
+        # Pack args in kwargs
+        kwargs.update(
+            {
+                "include_rescaling": include_rescaling,
+                "input_shape": input_shape,
+                "input_tensor": input_tensor,
+            }
+        )
+        return ResNetV2Backbone.from_preset("resnet34_v2", **kwargs)
 
+    @classproperty
+    def presets(cls):
+        """Dictionary of preset names and configurations."""
+        return {}
 
-def ResNet101V2(
-    *,
-    include_rescaling,
-    include_top,
-    classes=None,
-    weights=None,
-    input_shape=(None, None, 3),
-    input_tensor=None,
-    pooling=None,
-    classifier_activation="softmax",
-    name="resnet101v2",
-    **kwargs,
-):
-    """Instantiates the ResNet101V2 architecture."""
-    return ResNetV2(
-        stackwise_filters=MODEL_CONFIGS["ResNet101V2"]["stackwise_filters"],
-        stackwise_blocks=MODEL_CONFIGS["ResNet101V2"]["stackwise_blocks"],
-        stackwise_strides=MODEL_CONFIGS["ResNet101V2"]["stackwise_strides"],
-        name=name,
-        include_rescaling=include_rescaling,
-        include_top=include_top,
-        weights=weights,
-        input_shape=input_shape,
-        input_tensor=input_tensor,
-        pooling=pooling,
-        classes=classes,
-        classifier_activation=classifier_activation,
-        block_type="block",
+    @classproperty
+    def presets_with_weights(cls):
+        """Dictionary of preset names and configurations that include weights."""
+        return {}
+
+
+class ResNet50V2Backbone(ResNetV2Backbone):
+    def __new__(
+        self,
+        include_rescaling=True,
+        input_shape=(None, None, 3),
+        input_tensor=None,
         **kwargs,
-    )
+    ):
+        # Pack args in kwargs
+        kwargs.update(
+            {
+                "include_rescaling": include_rescaling,
+                "input_shape": input_shape,
+                "input_tensor": input_tensor,
+            }
+        )
+        return ResNetV2Backbone.from_preset("resnet50_v2", **kwargs)
+
+    @classproperty
+    def presets(cls):
+        """Dictionary of preset names and configurations."""
+        return copy.deepcopy(
```

Comment:
The copy call should be moved to right before `backbone_presets` (the rest of the dict gets generated on the fly) to avoid redundant work